tools/codegen/dest/register_dispatch_key.py

```diff
@@ -368,7 +368,7 @@ def generate_defn(cpp_sig: CppSignature) -> str:
 
                 device_guard = "// DeviceGuard omitted"  # default
                 if f.device_guard and self.backend_index.device_guard:
-                    has_tensor_options = any(isinstance(a.argument, TensorOptionsArguments) for a in args)
+                    has_tensor_options = any(isinstance(a, TensorOptionsArguments) for a in f.func.arguments.non_out)
```

Comment:
We should be looking at the `FunctionSchema` arguments to see if there are any TensorOptions args, not at the C++ kernel arguments. That's because in core, we use the `native` API to call the kernel (which includes `TensorOptions` while external dispatcher backends use the `dispatcher` API (which uses the expanded args instead).